### DISCUSSION TITLE
[General] VBS Deployment Script error at glpi-agent.vbs(406, 4) WinHttp.WinHttpRequest

### DISCUSSION THREAD

**Discussion by billgatos** [glpi-project/glpi-agent#357] [General]
### Your question

Hi all,
We have on our deployment the script to install the glpi-agent.
We got the latest VBS script from git

Were using the nightly builds.
We changed the vbs as supposed but something is wrong because we get error on line 406 WinHttp.WinHttpRequest

what have changed as today day 08/03/2023
SetupVersion = "1.5-git8529f898"
SetupNightlyLocation = "https://nightly.glpi-project.org/glpi-agent/"
SetupArchitecture = "x64" because there's no AUTO in github
SetupOptions = "/quiet RUNNOW=1 ADD_FIREWALL_EXCEPTION=1 NO_SSL_CHECK=1 RUNNOW=1 SSL_FINGERPRINT='sha256$................' SERVER='https://internal.company.pt/front/inventory.php'"
Setup = "GLPI-Agent-" & SetupVersion & "-" & SetupArchitecture & ".msi"
Repair = "Yes"

what could be the issue ?
Best regards.

**Comment by g-bougard** [comment]
Hi @billgatos,

first, you don't need to set `NO_SSL_CHECK` if you're using `SSL_FINGERPRINT`.

Second, when `SetupArchitecture` is set to "Auto", it means the script will detect the right arch for the computer. Nowadays, this means it will generally fix "Auto" by "x64" is the `Setup` var. But of course, you can set it to "x64" directly.

Then you should probably not set `Repair` to "Yes". Have you any reason to do it ?

Are you for the line number ? I don't see any WinHttp related code on the [current l.406](https://github.com/glpi-project/glpi-agent/blob/8529f898132e016ef0c304ce359a2b476aaaa71a/contrib/windows/glpi-agent-deployment.vbs#L406).

**Reply by billgatos** [reply]
Thanks for the repply.

The error happens in severall win11 but not on windows server 2016 and 2019
The idea of repair, was to upgrade the nighlty, over previous version.
The auto, is that we don't have any x86 already.

The line is 406 but i believe is not the error line, is when is called on the fuction itself

**Reply by g-bougard** [reply]
Okay, this is weird.
In principle, you don't need the `Repair` option. The installer always makes an "upgrade" if the full given version is different (this includes the -git* part).
Did you test it on windows 10 ? It seems the probably may be related to win11 now.

**Reply by g-bougard** [reply]
Also can describe with more details the context: how the script is started ? Has it sufficient privileges ? Where do you see the error ?
I tried the script on the Windows 11 dev evaluation image and it has no problem to download the last development build MSI file.

**Reply by billgatos** [reply]
going to test more and reply after.
Thanks once again

**Reply by billgatos** [reply]
This is my GPO
![imagem](https://user-images.githubusercontent.com/13196705/227220158-b86d60eb-5523-415f-b340-bdd6068032d6.png)
Inside the script i have:

SetupVersion = "1.5-git8529f898"
SetupLocation = "https://github.com/glpi-project/glpi-agent/releases/download/" & SetupVersion
SetupNightlyLocation = "https://nightly.glpi-project.org/glpi-agent/"
SetupArchitecture = "Auto"

SetupOptions = "/quiet RUNNOW=1 ADD_FIREWALL_EXCEPTION=1 NO_SSL_CHECK=1 RUNNOW=1 SSL_FINGERPRINT='sha256$blablabla' SERVER='https://myserver.pt/front/inventory.php'"

Setup = "GLPI-Agent-" & SetupVersion & "-" & SetupArchitecture & ".msi"

**Comment by billgatos** [comment]
Just tried again the scrip on a fresh windows 10 installation and the script gave the same error.

The autodeployment of glpi is done through a GPO at computer startup , so it has all the administrative access.
if manual i execute it on a command line with administrator privilege.

**Reply by g-bougard** [reply]
Can you resume what options you're using this time ? About the GPO, verify you're not using PowerShell.

**Reply by billgatos** [reply]
Sorry i replied above.

**Reply by g-bougard** [reply]
Okay,
as I said before, you don't need to set `NO_SSL_CHECK` if you're using `SSL_FINGERPRINT`. I see you're using 2 times `RUNNOW=1`, one time is sufficient.
Anyway, this shouldn't make any problem.
Also I remember I asked you `Where do you see the error ?` but got no response. This would help us to reproduce.

**Reply by billgatos** [reply]
hi here is the error.
sorry my fault
![image](https://user-images.githubusercontent.com/13196705/227468319-619ed008-be04-4a05-8baa-42dad2558d0d.png)

**Reply by billgatos** [reply]
Line 406

**Reply by billgatos** [reply]
My friend [g-bougard] problem solved, it's my super crazy antivirus from ESET , enterprise edition, problably doesn't like the way of getting the github updates.
turned off the firewall and no error.

**Reply by g-bougard** [reply]
Okay, that was the last thing I suspected.
Happy to figure out your problem.

**Reply by billgatos** [reply]
I have now to figure how am i going to accpet winhttp request in the eset firewall.....
best regards.
